datapipelines/src/test/java/io/customer/datapipelines/utils/Plugins.kt

```diff
@@ -0,0 +1,37 @@
+package io.customer.datapipelines.utils
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.IdentifyEvent
+import com.segment.analytics.kotlin.core.ScreenEvent
+import com.segment.analytics.kotlin.core.TrackEvent
+import com.segment.analytics.kotlin.core.platform.Plugin
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.core.util.Logger
+
+class OutputReaderPlugin(override var analytics: Analytics) : Plugin {
+    override val type: Plugin.Type = Plugin.Type.After
+
+    private val logger: Logger = SDKComponent.logger
+    val allEvents: MutableList<BaseEvent> = mutableListOf()
+
+    override fun execute(event: BaseEvent): BaseEvent {
+        allEvents.add(event)
+        logger.debug("[OutputReaderPlugin] Event type: ${event.type}, Event name: ${(event as? TrackEvent)?.event}")
+        return event
+    }
+
+    fun reset() {
+        allEvents.clear()
+    }
+}
+
+val OutputReaderPlugin.lastEvent: BaseEvent?
+    get() = allEvents.lastOrNull()
+
+val OutputReaderPlugin.identifyEvents: List<IdentifyEvent>
+    get() = allEvents.filterIsInstance<IdentifyEvent>()
+val OutputReaderPlugin.screenEvents: List<ScreenEvent>
+    get() = allEvents.filterIsInstance<ScreenEvent>()
+val OutputReaderPlugin.trackEvents: List<TrackEvent>
+    get() = allEvents.filterIsInstance<TrackEvent>()
```

Comment:
instead of repeating for every type and possibly adding more, would this be easier? open to keeping the current as well, if that improves readability. 

```suggestion
inline fun <reified T : BaseEvent> getEventsOfType(): List<T> {
        return allEvents.filterIsInstance<T>()
    }
```

Reply:
I think it just kills the concept of these extensions. I added this to get rid of specifying classes. Otherwise, `plugin.allEvents.filterIsInstance<TrackEvent>` and `plugin.getEventsOfType<TrackEvent>()` is almost same to me and `getEventsOfType` looks unnecessary here. So I'll stick to current implementation for ease of casting and not specifying these classes every time.